rellic/AST/Z3ConvVisitor.cpp

```diff
@@ -794,14 +797,31 @@ void Z3ConvVisitor::VisitUnaryApp(z3::expr z_op) {
       break;
 
     case Z3_OP_EXTRACT: {
-      CHECK(t_sub->isIntegerType()) << "Extract operand is not an integer";
-      auto s_size{GetZ3SortSize(z_op.get_sort())};
-      auto t_sign{t_sub->isSignedIntegerType()};
-      auto &ti{ast_ctx->getTargetInfo()};
-      auto t_size{ti.getTypeWidth(ti.getLeastIntTypeByWidth(s_size, t_sign))};
-      auto t_op{ast_ctx->getIntTypeForBitwidth(t_size, t_sign)};
-      c_op = CreateCStyleCastExpr(*ast_ctx, t_op,
-                                  clang::CastKind::CK_IntegralCast, c_sub);
+      auto t_uint{ast_ctx->UnsignedIntTy};
+      auto t_uint_size{ast_ctx->getTypeSize(t_uint)};
+      auto t_res{ast_ctx->getIntegerTypeOrder(t_sub, t_uint) < 0 ? t_uint
+                                                                 : t_sub};
+      // Shift value
```

Comment:
Can you change these comments to describe the higher level transformation you're doing? Somthing like, `Given a `(extract blah A B)` produce an expression of the form `(blah >> foo) & bar`." Right now the commenting here provides no additional value over your variable names.